app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,28 @@
+class SessionsController < ApplicationController
+  #this line is for the omniauth developer strategy login.
+  skip_before_filter :verify_authenticity_token, only: :create
+  skip_before_action :require_login, only: [:new, :create, :destroy]
+
+  def create
+    #can sign in with twitter or vimeo
+    auth_hash = request.env['omniauth.auth']
+    if auth_hash["uid"]
+      @user = User.find_or_create_from_omniauth(auth_hash)
+      if @user
+        session[:user_id] = @user.id
+      else
+        flash[:notice] = "Failed to save the user"
+      end
```

Comment:
Do you need to write this as a nested `if`-block, or is there a different way to do this?